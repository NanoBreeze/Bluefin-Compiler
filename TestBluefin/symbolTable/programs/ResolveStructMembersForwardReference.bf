struct First {
	int a = 1;
	
	void f(int a) {
		int x = g();
		bool b = h();
	}
	
	int g() {
		return 2;
	}
	
	bool h() {
		return b;
	}
	bool b;
};