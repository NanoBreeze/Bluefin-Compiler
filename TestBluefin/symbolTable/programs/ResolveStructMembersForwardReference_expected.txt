declare - First - c_struct - t_First
setCurrentScope - Level 1
declare - a - c_var - t_int
declare - f - c_function - t_void
enterScope - Level 2
declare - a - c_var - t_int
declare - x - c_var - t_int
declare - b - c_var - t_bool
exitScope - Level 2
declare - g - c_function - t_int
enterScope - Level 2
exitScope - Level 2
declare - h - c_function - t_bool
enterScope - Level 2
exitScope - Level 2
declare - b - c_var - t_bool
exitScope - Level 1
resolve - int - c_builtinType - t_int
resolve - void - c_builtinType - t_void
resolve - int - c_builtinType - t_int
resolve - int - c_builtinType - t_int
resolve - g - c_function - t_int
resolve - bool - c_builtinType - t_bool
resolve - h - c_function - t_bool
resolve - int - c_builtinType - t_int
resolve - bool - c_builtinType - t_bool
resolve - b - c_var - t_bool
resolve - bool - c_builtinType - t_bool
